caikit/core/model_management/model_saver.py

```diff
@@ -0,0 +1,117 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file defines the abstraction for handling the output of training a model"""
+# Standard
+from typing import Optional
+import abc
+import os
+import typing
+
+# Local
+from ..modules import ModuleBase
+from caikit.interfaces.common.data_model.stream_sources import PathReference
+
+T = typing.TypeVar("T")
+
+
+class ModelSaver(typing.Generic[T]):
+    """Generic-typed, abstract base for model saver"""
+
+    @abc.abstractmethod
+    def save_model(
+        self, model: ModuleBase, model_name: str, training_id: Optional[str]
+    ) -> typing.Any:
```

Comment:
Potentially this is just `T` (i.e. saver for `str` path, saver for `s3Path`, etc...)